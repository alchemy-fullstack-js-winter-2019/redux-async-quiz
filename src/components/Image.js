import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Image extends PureComponent {
  static propTypes = {
    img: PropTypes.string.isRequired,
    fetchDog: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.fetchDog();
  }

  render() {
    const { img } = this.props;
    return (
      <p>{img}</p>
    );
  }

}

export default Image;




