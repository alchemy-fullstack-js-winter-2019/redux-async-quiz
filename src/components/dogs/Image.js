import React from 'react';
import PropTypes from 'prop-types';
import Load from './Load';

const Image = ({ dogImage, fetch }) => {
  return (
    <div>
      <Load fetch={fetch}/>
      <br/>
      <img src={dogImage}/>
    </div>
  );
};

Image.propTypes = {
  dogImage: PropTypes.string.isRequired,
  fetch: PropTypes.func.isRequired
};

export default Image;
