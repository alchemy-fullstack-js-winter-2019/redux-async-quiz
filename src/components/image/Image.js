import React from 'react';
import PropTypes from 'prop-types';

export default function Image({ dogImage }) {
  return (
    <>
      <img src={dogImage} />
    </>
  );
}

Image.propTypes = {
  dogImage: PropTypes.object.isRequired
};
