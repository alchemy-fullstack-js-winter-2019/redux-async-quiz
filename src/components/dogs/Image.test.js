import React from 'react';
import { shallow } from 'enzyme';
import Image from './Image';

describe('Image', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Image dogImage={'https://images.dog.ceo/breeds/cairn/n02096177_599.jpg'} 
        fetch={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
