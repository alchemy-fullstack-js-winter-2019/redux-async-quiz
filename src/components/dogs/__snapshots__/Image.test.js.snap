// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches a snapshot 1`] = `
<div>
  <Load
    fetch={[MockFunction]}
  />
  <br />
  <img
    src="https://images.dog.ceo/breeds/cairn/n02096177_599.jpg"
  />
</div>
`;
